### ISSUE TITLE
persist.sql transaction test cases are failing with change of `transactionInfoMap` to ConcurrentHashMap

### ISSUE THREAD

**Issue by kaneeldias** [ballerina-platform/persist-tools#318]
**Description:**
The following change in lang results in the failure of the transaction test cases in `persist.sql`
https://github.com/ballerina-platform/ballerina-lang/pull/41506

```
[fail] mssqlTransactionTest:

		    error {ballerina/test:0}TestError ("NotFoundError expected")
				callableName: createBallerinaError moduleName: ballerina.test.0 fileName: assert.bal lineNumber: 41
				callableName: assertTrue moduleName: ballerina.test.0 fileName: assert.bal lineNumber: 61
				callableName: mssqlTransactionTest moduleName: ballerinax.persist.sql$test.1.tests.mssql-transaction-tests fileName: tests/mssql-transaction-tests.bal lineNumber: 37
				callableName: mssqlTransactionTest$lambda78$ moduleName: ballerinax.persist.sql$test.1.tests.test_execute-generated_1 fileName: tests/test_execute-generated_1.bal lineNumber: 82
			

		[fail] mysqlTransactionTest:

		    error {ballerina/test:0}TestError ("NotFoundError expected")
				callableName: createBallerinaError moduleName: ballerina.test.0 fileName: assert.bal lineNumber: 41
				callableName: assertTrue moduleName: ballerina.test.0 fileName: assert.bal lineNumber: 61
				callableName: mysqlTransactionTest moduleName: ballerinax.persist.sql$test.1.tests.mysql-transaction-tests fileName: tests/mysql-transaction-tests.bal lineNumber: 37
				callableName: mysqlTransactionTest$lambda170$ moduleName: ballerinax.persist.sql$test.1.tests.test_execute-generated_1 fileName: tests/test_execute-generated_1.bal lineNumber: 175
```

The issue seems to be with the transaction context not being maintained when executing the query methods.

**Comment by gimantha** [comment]
@dsplayerX is currently working on a fix. It should fix this problem from ballerina-lang repo

**Comment by dsplayerX** [comment]
Fixed with commit [20d0438](https://github.com/ballerina-platform/ballerina-lang/pull/41506/commits/20d0438e2fed9797383430882e800d48dd4fbbf0) in https://github.com/ballerina-platform/ballerina-lang/pull/41506.
All tests are passing now.
